nav2_common/cmake/nav2_package.cmake

```diff
@@ -42,4 +42,16 @@ macro(nav2_package)
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} --coverage")
     set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} --coverage")
   endif()
+
+  # Defaults for Microsoft C++ compiler
+  if(MSVC)
```

Comment:
You should make a ROS2 tutorial on ros2_documentation to explain how to do this simply. Also posting on Discourse to tell people about this so that we can have a scalable support for Windows. 

Can you do a once over of the ROS2 ecosystem (ros2/*, ros-* etc) and file tickets about this change? This is a BIG improvement!